### PR TITLE
Add LRU caching to get task method

### PR DESCRIPTION
**Pull request by owlishDeveloper**
Bugzilla Bug: [1595221](https://bugzilla.mozilla.org/show_bug.cgi?id=1595221)

Github Bug/Issue: Fixes #1595221

### CODE REVIEW

File: services/queue/src/main.js [hunk 1]
```diff
@@ -431,6 +432,7 @@ let load = loader({
         blobRegion: ctx.cfg.app.blobArtifactRegion,
         publicBlobBucket: ctx.cfg.app.publicBlobArtifactBucket,
         privateBlobBucket: ctx.cfg.app.privateBlobArtifactBucket,
+        LRUcache: new QuickLRU({maxSize: 1000000}),
```
Comment (djmitche) [inline services/queue/src/main.js]: Let's make this a configurable value, with a small default (10?).  At a conservative 3k apiece, a million Task objects would occupy 3GB, which is more than we budget for these k8s pods.

File: services/queue/config.yml [hunk 2]
```diff
@@ -100,6 +100,8 @@ defaults:
     useCloudMirror:               !env:bool:optional USE_CLOUD_MIRROR
     cloudMirrorHost:              'cloud-mirror.taskcluster.net'
 
+    taskCacheMaxSize: 5000
```
Comment (djmitche) [inline services/queue/config.yml]: To make this changeable without releasing a new TC, it needs to be an !env:.. variable.  I think that can use `optional` too, with a suitable default value.

**Comment by imbstack** [comment]
I wouldn't say I have any edge cases in mind in particular. It feels like the kind of thing that we should get 100% right is all I mean. I'll look at this today.

File: services/queue/src/api.js [hunk 3]
```diff
@@ -142,19 +143,23 @@ builder.declare({
...
+  const {taskId} = req.params;
+
+  // Load Task entity if it's absent from cache
+  let task;
+  if (this.LRUcache.has(taskId)) {
+    task = this.LRUcache.get(taskId);
+  } else {
+    task = await this.Task.load({taskId}, true);
+    this.LRUcache.set(taskId, task);
+  }
```
Comment (imbstack) [inline services/queue/src/api.js]: I think we need this sorta thing in other places where we load tasks such as `claimWork`

Comment (djmitche) [reply services/queue/src/api.js]: That's not a bad idea, but let's start small with the caching and see what the impact on our load is.

Comment (imbstack) [reply services/queue/src/api.js]: I don't think much of our load is coming from calls to `task` itself? I don't mind releasing this as-is though.

File: changelog/bug-1595221.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+level: patch
+reference: bug 1595221
+---
+Adds an LRU cache to getTask method, so that we don't have to make too many calls to Azure (tasks are immutable anyways)
+The default value for the cache size is 10. The name of the optional prop in the config.yml is `taskCacheMaxSize`
```
Comment (imbstack) [inline changelog/bug-1595221.md]: It should be `queue.task_cache_max_size` in this case.

**Comment by imbstack** [comment]
Although I do wonder if the queue test failure is "real". I think your code is correct but the test might be asserting something that is broken by this change

**Comment by owlishDeveloper** [comment]
> Although I do wonder if the queue test failure is "real". I think your code is correct but the test might be asserting something that is broken by this change

Which queue test failure?

**Pull request merged**